src/adaptermanager.js

```diff
@@ -211,6 +214,65 @@ exports.makeBidRequests = function(adUnits, auctionStart, auctionId, cbTimeout,
   return bidRequests;
 };
 
+function isValidBidRequest(adUnits) {
+  adUnits.forEach(adUnit => {
+    if (adUnit.sizes) {
+      utils.logWarn('Usage of adUnits.sizes will eventually be deprecated.  Please define size dimensions within the corresponding area of the mediaTypes.<object> (eg mediaTypes.banner.sizes).');
+    }
+
+    if (adUnit.mediaTypes) {
```

Comment:
generally recommend combining conditions to reduce the number of nested `if` conditions. For example you could combine into ` if (adUnit.mediaTypes && adUnit.mediaTypes.banner)`

Also since you are doing a lot of testing of on `adUnit.mediaTypes` - you can also assign this to a var:
`const mediaTypes = adUnit.mediaTypes;` and that will shorten the conditions:
`if (mediaTypes && mediaTypes.banner) {...}`

Reply:
Hey @mkendall07 In regards to the condition breakdown, I made them so heavily layered to avoid have the checks run when they shouldn't.  As an example if I were to combine `adUnit.mediaTypes && adUnit.mediaTypes.banner), this condition would fail for anyone trying to serve a video or native ad (as the banner doesn't exist) and they would see the subsequent error/warning messages in the console log for something they weren't intentionally using.  The same applied when I tried to combine some of the inner checks with the parent object (eg adUnit.mediaTypes.native.image && adUnit.mediaTypes.native.image.size).  If they were to intentionally not use the image asset, they would get the warning message because half the condition would fail.

If there's another way to handle this type of setup while still supporting ideal failure conditions, please let me know.  

I'll look into establishing variables though to help lessen the length of these statements.

Reply:
Had a talk with Matt separately.  Will make some changes where applicable.